app/services/katello/event_queue.rb

```diff
@@ -0,0 +1,38 @@
+module Katello
+  class EventQueue
+    @event_types = {}
+
+    def self.clear_events(event_type, object_id, on_or_earlier_than)
+      Katello::Event.where(:in_progress => true, :object_id => object_id, :event_type => event_type).where('created_at <= ?', on_or_earlier_than).destroy_all
+    end
+
+    def self.peek_event
```

Comment:
Do I read it correct that this function is setting all events that match the top of queue to in progress? If yes, I would not expect that a function named "peek" to change any date since it implies a viewing rather than modifying type of action.